packages/neptune-extension/src/common/components/upload-modal/UploadForm.tsx

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { NotebookDTO } from 'generated/leaderboard-client/src/models';
+
+import {
+  PlatformNotebook,
+  PlatformNotebookMetadata,
+} from 'types/platform';
+
+import {
+  uploadNotebook,
+  uploadCheckpoint,
+  getNotebook,
+} from 'common/utils/upload';
+
+import { PROJECT_LOCAL_STORAGE_KEY } from 'common/utils/localStorage';
+
+import Modal from 'common/components/modal/Modal';
+import ProjectInput from 'common/components/input/ProjectInput';
+
+import { getConfigurationState } from 'common/state/configuration/selectors';
+import { getNotebookState } from 'common/state/notebook/selectors'
+import { setNotebook } from 'common/state/notebook/actions'
+
+interface UploadFormProps {
+  platformNotebook: PlatformNotebook
+  onClose: () => void
+}
+
+type UploadMode = 'notebook' | 'checkpoint'
+
+const UploadForm: React.FC<UploadFormProps> = ({
```

Comment:
This file is too big. I would suggest encapsulate logic in one file and render in another stateless 'stupid' component.

Reply:
Excuse me, but by which standard is it too big ? It is not uncommon to see files with >1K lines in projects.

But to the point, I am not really fond of this distinction between "logic" and "plain" components because if you display a notice in html based on some prop it is logic. period. Some people try to call it "app logic" and "ui logic", as this discussion predates react. But to me it is better to properly structure the code than to introduce such notions, and in this particular example I cannot think on any other abstractions I can create, so I prefer having in all together and being sure it holds together, that implementing poor abstractions that no one understands

I still cannot grasp how a division of this component into two would help, and beside that you have to pass a ton of unrelated props that make only sense when seen together.

If you have a proposal on a specific abstraction I can create, please write it, but unless this compoment grows or we introduce a form library, this holds in my opinion.